components/brave_news/browser/publishers_controller.cc

```diff
@@ -262,6 +262,33 @@ void PublishersController::EnsurePublishersIsUpdating(
                                 .timeout = GetDefaultRequestTimeout()});
 }
 
+std::string PublishersController::GetFallbackLocale(const std::string& requested_locale, const base::flat_set<std::string>& available_locales) {
+
+  // Exact match
+  if (available_locales.find(requested_locale) != available_locales.end()) {
+    return requested_locale;
+  }
+
+  // Regional fallback
+  std::string country_code = requested_locale.substr(requested_locale.find('_') + 1);
+  for (const auto& locale : available_locales) {
+    if (locale.substr(locale.find('_') + 1) == country_code) {
```

Comment:
how do we enforce priority? For example, when `en_US`, `es_US` are both there but the user has `it_US`? We should have a regional default, i.e. `en_US`

Reply:
Hmm, do you think this is that useful? Say I only speak Spanish, so my locale is `es_US` but we only have `en_US`, I'm not going to be thrilled that I'm getting a feed in English.

More concretely, I lived in Poland for a while, but I didn't speak much Polish, so my locale was `en_PL` - a feed in Polish wouldn't have been that useful for me.